pkgs/development/compilers/eunomia-bpf/eunomia-cc/default.nix

```diff
@@ -0,0 +1,96 @@
+{ lib
+, stdenv
+, makeWrapper
+, fetchFromGitHub
+, rustPlatform
+, pkg-config
+, llvmPackages_16
+, elfutils
+, zlib
+}:
+let
+  bpftool = stdenv.mkDerivation {
+    pname = "bpftool";
+    version = "unstable-2023-03-11";
+    src = fetchFromGitHub {
+      owner = "eunomia-bpf";
+      repo = "bpftool";
+      rev = "05940344f5db18d0cb1bc1c42e628f132bc93123";
+      hash = "sha256-g2gjixfuGwVnFlqCMGLWVPbtKOSpQI+vZwIZciXFPTc=";
+      fetchSubmodules = true;
+    };
+
+    buildInputs = [ llvmPackages_16.clang llvmPackages_16.libllvm.dev elfutils zlib ];
+
+    buildPhase = ''
+      make -C src
+    '';
+
+    installPhase = ''
+      # dir structure compatible with `build.rs` from upstream
+      mkdir -p $out/src/libbpf
+      cp -r src/libbpf/include $out/src/libbpf
+      cp src/bpftool $out/src
+    '';
+  };
+
+  vmlinux-headers = fetchFromGitHub {
+    owner = "eunomia-bpf";
+    repo = "vmlinux";
+    rev = "933f83becb45f5586ed5fd089e60d382aeefb409";
+    hash = "sha256-CVEmKkzdFNLKCbcbeSIoM5QjYVLQglpz6gy7+ZFPgCY=";
+  };
+
+in
+rustPlatform.buildRustPackage rec {
+
+  pname = "ecc";
+  version = "1.0.11";
+
+  src = fetchFromGitHub {
+    owner = "eunomia-bpf";
+    repo = "eunomia-bpf";
+    rev = "v${version}";
+    hash = "sha256-UiwS+osyC3gtbQH0bWNsx1p3xYr993/FAZ5d5NKnaBM=";
+  };
+
+  cargoRoot = "compiler/cmd";
+
+  cargoLock.lockFile = ./Cargo.lock;
+
+  nativeBuildInputs = [
+    pkg-config
+    makeWrapper
+  ] ++ (with rustPlatform; [
+    bindgenHook
+  ]);
+
+  buildInputs = [
+    elfutils
+    zlib
+  ];
+
+  preBuild = ''
+    export SANDBOX=1
+    export OUT_DIR=$(pwd)
+    export VMLINUX_DIR=${vmlinux-headers}
+    export BPFTOOL_DIR=${bpftool}
+    cd ${cargoRoot}
+  '';
+
+  # most tests will try creating files out of the sandbox or accessing networks
```

Comment:
If it is possible to control the location of the files outside the sandbox via an environment variable, like `$HOME`, maybe it'd be worth trying that out? Maybe try out something similar to [this checkFlags](https://nixos.org/manual/nixpkgs/unstable/#disabling-package-tests) to disable the tests that try to access the network.

Reply:
> If it is possible to control the location of the files outside the sandbox via an environment variable, like `$HOME`, maybe it'd be worth trying that out? Maybe try out something similar to [this checkFlags](https://nixos.org/manual/nixpkgs/unstable/#disabling-package-tests) to disable the tests that try to access the network.

`$HOME` has been set to pass some tests, but others require dynamic link `libclang` or clang binary which are not found in check env. I have no idea how to solve this now.

Tried exporting `LIBCLANG_PATH` and `PATH` in `preCheck` phase but none works. 

This will not become a problem at runtime since it is wrapped.